azkaban-common/src/main/java/azkaban/utils/OsCpuUtil.java

```diff
@@ -0,0 +1,195 @@
+    final long startTimeCpuIdle = startCpuStats.getTimeCpuIdle();
+    final long endSysUptime = endCpuStats.getSysUptime();
+    final long endTimeCpuIdle = endCpuStats.getTimeCpuIdle();
+
+    if (endSysUptime == startSysUptime) {
+      logger.error("Failed to calculate cpu load: division by zero");
+      return -1.0;
+    }
+    final double percentageCpuIdle =
+        (100.0 * (endTimeCpuIdle - startTimeCpuIdle)) / (endSysUptime - startSysUptime);
+    return 100.0 - percentageCpuIdle;
+  }
+
+  private CpuStats getCpuStats() {
+    if (!Files.isRegularFile(Paths.get(CPU_STAT_FILE))) {
+      // Mac doesn't use proc pseudo files for example.
+      return null;
+    }
+
+    final String cpuLine = getCpuLineFromStatFile();
+    if (cpuLine == null) {
+      return null;
+    }
+
+    return getCpuStatsFromLine(cpuLine);
+  }
+
+  private String getCpuLineFromStatFile() {
+    BufferedReader br = null;
+    try {
+      br = Files.newBufferedReader(Paths.get(CPU_STAT_FILE), StandardCharsets.UTF_8);
+      String line;
+      while ((line = br.readLine()) != null) {
+        // looking for a line starting with "cpu<space>" which aggregates the values in all of
+        // the other "cpuN" lines.
+        if (line.startsWith("cpu ")) {
+          return line;
+        }
+      }
+    } catch (final IOException e) {
+      final String errMsg = "Failed to read cpu stat file: " + CPU_STAT_FILE;
+      logger.error(errMsg, e);
+    } finally {
+      if (br != null) {
+        try {
+          br.close();
+        } catch (final IOException e) {
+          final String errMsg = "Failed to close cpu stat file: " + CPU_STAT_FILE;
+          logger.error(errMsg, e);
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Parses cpu usage information from /proc/stat file.
+   * Example of line expected with the meanings of the values below:
+   * cpu  4705  356  584    3699   23    23     0       0     0          0
+   * ---- user nice system idle iowait  irq  softirq steal guest guest_nice
+   *
+   * Method visible within the package for testing purposes.
+   *
+   * @param line the text containing cpu usage statistics
+   * @return CpuStats object. null if there is an error.
+   */
+  CpuStats getCpuStatsFromLine(final String line) {
+    try {
+      final String[] cpuInfo = line.split("\\s+");
+      final long user = Long.parseLong(cpuInfo[1]);
+      final long nice = Long.parseLong(cpuInfo[2]);
+      final long system = Long.parseLong(cpuInfo[3]);
+      final long idle = Long.parseLong(cpuInfo[4]);
+      final long iowait = Long.parseLong(cpuInfo[5]);
+      final long irq = Long.parseLong(cpuInfo[6]);
+      final long softirq = Long.parseLong(cpuInfo[7]);
+      final long steal = Long.parseLong(cpuInfo[8]);
+
+      // guest and guest_nice are counted on user and nice respectively, so don't add them
+      final long totalCpuTime = user + nice + system + idle + iowait + irq + softirq + steal;
```

Comment:
Consider adding the link to reference the way of calculating cpu time?

Reply:
will add it.